src/database.js

```diff
@@ -1280,6 +1281,70 @@ Database.prototype.runTransaction = function(options, runFn) {
   });
 };
 
+/**
+ * A function to execute in the context of a transaction.
+ * @param {Transaction} transaction The transaction object. The transaction has
+ *     already been created, and is ready to be queried and committed against.
+ */
+/**
+ * A transaction in Cloud Spanner is a set of reads and writes that execute
+ * atomically at a single logical point in time across columns, rows, and tables
+ * in a database.
+ *
+ * Note that Cloud Spanner does not support nested transactions. If a new
+ * transaction is started inside of the run function, it will be an independent
+ * transaction.
+ *
+ * The async function you provide will become the "run function". It
+ * will be executed with a {@link Transaction}
+ * object. The Transaction object will let you run queries and queue mutations
+ * until you are ready to {@link Transaction#commit}.
+ *
+ * In the event that an aborted error occurs, we will re-run the `runFn` in its
+ * entirety. If you prefer to handle aborted errors for yourself please refer to
+ * {@link Database#getTransaction}.
+ *
+ * For a more complete listing of functionality available to a Transaction, see
+ * the {@link Transaction} API documentation. For a general overview of
+ * transactions within Cloud Spanner, see
+ * [Transactions](https://cloud.google.com/spanner/docs/transactions) from the
+ * official Cloud Spanner documentation.
+ *
+ * @see [Transactions](https://cloud.google.com/spanner/docs/transactions)
+ * @see [Timestamp Bounds](https://cloud.google.com/spanner/docs/timestamp-bounds)
+ *
+ *
+ * @example
+ * const Spanner = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ *
+ * const instance = spanner.instance('my-instance');
+ * const database = instance.database('my-database');
+ *
+ * await database.runTransactionAsync(async (transaction) => {
+ *   const [rows] = await transaction.run('SELECT * FROM MyTable');
+ *   const data = rows.map(row => row.thing);
+ *
+ *   await transaction.commit();
+ *   return data;
+ * }).then(data => {
+ *   // ...
+ * });
+ */
+Database.prototype.runTransactionAsync = function(runFn) {
```

Comment:
You don't have to use `async` here :) This could be re written as something like:

```js
Database.prototype.runTransactionAsync = function(runFn) {
 return retry(
    this.getTransaction().then(r => {
      const transaction = r[0];
      return runFn(transaction);
    }).catch(e => {
      if (e.code === ABORTED) throw e;
      throw new retry.AbortError(e.message);
    });
 );
}
```